src/transports/smart_protocol.c

```diff
@@ -236,13 +236,14 @@ static int recv_pkt(git_pkt **out, gitno_buffer *buf)
 	} while (error);
 
 	gitno_consume(buf, line_end);
-	pkt_type = pkt->type;
+	if (pkt_type)
+		*pkt_type = pkt->type;
 	if (out != NULL)
 		*out = pkt;
 	else
 		git__free(pkt);
 
-	return pkt_type;
+	return error;
```

Comment:
:+1: This is our preferred style and definitly easier to handle than the previous interface